bundles/org.openhab.binding.touchwand/src/main/java/org/openhab/binding/touchwand/internal/TouchWandAlarmSensorHandler.java

```diff
@@ -0,0 +1,130 @@
+
+import static org.openhab.binding.touchwand.internal.TouchWandBindingConstants.*;
+
+import java.util.Iterator;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.touchwand.internal.dto.TouchWandAlarmSensorCurrentStatus.Sensor;
+import org.openhab.binding.touchwand.internal.dto.TouchWandAlarmSensorCurrentStatus.bSensorEvent;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitData;
+import org.openhab.binding.touchwand.internal.dto.TouchWandUnitDataAlarmSensor;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link TouchWandAlarmSensorHandler} is responsible for handling command for Alarm Sensor unit
+ *
+ *
+ * @author Roie Geron - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class TouchWandAlarmSensorHandler extends TouchWandBaseUnitHandler {
+
+    private static final int BATT_LEVEL_LOW = 20;
+    private static final int BATT_LEVEL_LOW_HYS = 5;
+
+    private boolean isBatteryLow = false;
+
+    public TouchWandAlarmSensorHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    void updateTouchWandUnitState(TouchWandUnitData unitData) {
+        if (unitData instanceof TouchWandUnitDataAlarmSensor) {
+            updateBatteryLevel((TouchWandUnitDataAlarmSensor) unitData);
+            updateIllumination((TouchWandUnitDataAlarmSensor) unitData);
+            updateChannelLeak((TouchWandUnitDataAlarmSensor) unitData);
+            updateChannelDoorWindow((TouchWandUnitDataAlarmSensor) unitData);
+            updateChannelMotion((TouchWandUnitDataAlarmSensor) unitData);
+            updateChannelTemprature((TouchWandUnitDataAlarmSensor) unitData);
+        } else {
+            logger.warn("updateTouchWandUnitState incompatible TouchWandUnitData instance");
+        }
+    }
+
+    @Override
+    void touchWandUnitHandleCommand(Command command) {
+    }
+
+    void updateBatteryLevel(TouchWandUnitDataAlarmSensor unitData) {
+        Integer battLevel = unitData.getCurrStatus().getBatt();
+        updateState(CHANNEL_BATTERY_LEVEL, new DecimalType(battLevel));
+        int lowThreshold = isBatteryLow ? BATT_LEVEL_LOW + BATT_LEVEL_LOW_HYS : BATT_LEVEL_LOW;
+        boolean lowBattery = (battLevel <= lowThreshold);
+        updateState(CHANNEL_BATTERY_LOW, OnOffType.from(lowBattery));
+        isBatteryLow = (battLevel <= lowThreshold);
+    }
+
+    void updateIllumination(TouchWandUnitDataAlarmSensor unitData) {
+        Iterator<Sensor> iter = unitData.getCurrStatus().getSensorsStatus().iterator();
+        while (iter.hasNext()) {
+            Sensor mySensor = iter.next();
+            if (mySensor.type == SENSOR_TYPE_LUMINANCE) {
+                updateState(CHANNEL_ILLUMINATION, new DecimalType(mySensor.value));
+            }
+        }
```

Comment:
Please don't use Iterator unless you are going to take advantage of `remove()`.
I suggest using either for-each or streaming here instead.

For-each:
```suggestion
        for(Sensor sensor : unitData.getCurrStatus().getSensorsStatus()){
            if (sensor.type == SENSOR_TYPE_LUMINANCE) {
                updateState(CHANNEL_ILLUMINATION, new DecimalType(sensor.value));
            }
        }
```
Streaming:
```suggestion
        unitData.getCurrStatus().getSensorsStatus().stream()
            .filter(sensor -> sensor.type == SENSOR_TYPE_LUMINANCE)
            .forEach(sensor -> updateState(CHANNEL_ILLUMINATION, new DecimalType(sensor.value)));
```

Please update the other methods as well.